anton_gadzikovskiy/graph_generator.cpp

```diff
@@ -107,46 +206,168 @@ string print_graph(const Graph& graph) {
   graph_output += "]\n}\n";
   return graph_output;
 }
+
 }  // namespace graph_printing
 
-Graph generate_graph() {
-  auto graph = Graph();
-
-  for (int i = 0; i < kVerticesCount; i++) {
-    graph.add_vertex();
-  }
-
-  graph.add_edge(0, 1);
-  graph.add_edge(0, 2);
-  graph.add_edge(0, 3);
-  graph.add_edge(1, 4);
-  graph.add_edge(1, 5);
-  graph.add_edge(1, 6);
-  graph.add_edge(2, 7);
-  graph.add_edge(2, 8);
-  graph.add_edge(3, 9);
-  graph.add_edge(4, 10);
-  graph.add_edge(5, 10);
-  graph.add_edge(6, 10);
-  graph.add_edge(7, 11);
-  graph.add_edge(8, 11);
-  graph.add_edge(9, 12);
-  graph.add_edge(10, 13);
-  graph.add_edge(11, 13);
-  graph.add_edge(12, 13);
-
-  return graph;
+bool is_generated(double prob) {
+  srand(time(0));
+  return (rand() % 100) <= prob;
+}
+
+const VertexId& get_random_vertex_id(const vector<VertexId>& vertices) {
+  srand(time(0));
+  const int random_id = rand() % vertices.size();
+  return vertices[random_id];
+}
+
+void generate_gray_edges(Graph& graph, int depth, int new_vertices_num) {
+  graph.add_vertex();
+  VertexId new_child_vertex_id = 1;
+  for (int i = 0; i < depth; i++) {
```

Comment:
Названия не достаточно очевидные:
```suggestion
  for (int current_depth = 0; depth < depth; current_depth++) {
```